rfc-183-publishing-flexible-pages.md

```diff
@@ -0,0 +1,162 @@
+    - name: content
+      type: govspeak
+      required: true
+```
+
+In this design, the schema is a list of named sections, each with fixed subsections. It is expected that each section would take up the entire width of the screen on the frontend.
+
+We are also considering a second possible design, which specifies a layout. The same history page example above would look like this in the second design:
+
+```yaml
+document_type: history_page
+header:
+  - section: page_title
+    uid: title
+    inputs:
+      - name: heading_text
+        type: text
+        required: true
+      - name: context
+        type: text
+      - name: lead_paragraph
+        type: text
+sidebar:
+  - section: image
+    uid: sidebar_image
+    inputs:
+      - name: image
+        inputs:
+          - name: source
+            type: url
+            required: true
+          - name: alt
+            type: text
+          - name: caption
+            type: text
+  - section: contents_list
+    uid: page_contents
+    inputs:
+      - name: items
+        children:
+          - name: href
+            type: text
+            required: true
+          - name: text
+            type: text
+            required: true
+main_content:
+  - section: rich_text
+    uid: body
+    inputs:
+      - name: content
+        type: govspeak
+        required: true
+```
+
+In this design the schema defines a number of page regions which can be populated by one or more sections. This may allow for easier changes to flexible page layouts, where we are just re-arranging existing components. We would be able to make some changes without involving the frontend teams.
+
+#### Decision
+
+### 2. How strict should the Publishing API content schema be in enforcing the content?
+
+There are two simple approaches to validating the flexible page content sent to Publishing API:
+
+1. No validation of the flexible page content
+2. Write a schema for each flexible page type specifying exactly what attributes should be present within the content
+
+If we were to adopt the second design from decision 1, which places sections within a particular region, then an additional approach is possible:
+
+3. Write a single schema for all flexible page types which validates the individual sections, but allows for any sections to be used in any region.
+
+#### Decision
+
+### 3. Should the publishing interface for flexible pages be built in Whitehall, Specialist Publisher, or another application?
+
+From our existing publishing applications, there are two obvious candidates to host flexible page publishing: Whitehall and Specialist Publisher.
+
+Factors in favour of hosting flexible page publishing in Whitehall:
+
+1. Whitehall has the most mature feature set. Extending the Whitehall edition model would enable flexible pages with the full publishing workflow, including scheduling and 2i.
+2. Whitehall is the most commonly used publishing tool, so publishers will not need to adapt to a new tool.
```

Comment:
I wonder if there's an additional benefit to Whitehall, in that it has fairly robust support for assets (e.g. uploading images, cropping them etc.) which AFAIK specialist doesn't have.

I imagine assets is going to be a big part of this whole shindig.